docs/book/src/migration/plugin/v2_v3.md

```diff
@@ -1,3 +1,570 @@
+domain: my.domain
+layout: go.kubebuilder.io/v3
+projectName: example
+repo: example
+resources:
+- api:
+    crdVersion: v1beta
+  group: webapp
+  kind: Guestbook
+  version: v1
+  webhooks:
+    webhookVersion: v1beta
+version: 3-alpha
+```
+
+### Upgrade the Go version and its dependencies:  
+
+Ensure that your `go.mod` is using the go version `1.15` as the following dependencies versions:
+
+```
+go 1.15
+
+require (
+	github.com/go-logr/logr v0.3.0
+	github.com/onsi/ginkgo v1.14.1
+	github.com/onsi/gomega v1.10.2
+	k8s.io/api v0.19.2
+	k8s.io/apimachinery v0.19.2
+	k8s.io/client-go v0.19.2
+	sigs.k8s.io/controller-runtime v0.7.0
+)
+```
+
+### Update the golang image 
+
+In th Dockerfile, replace:
+
+```
+# Build the manager binary
+FROM golang:1.13 as builder
+```
+
+With:
+```
+# Build the manager binary
+FROM golang:1.15 as builder
+```
+
+### Update your Makefile
+
+To allow the tool scaffold the new API versions and use properly the new controller-gen, replace:
+
+```
+CRD_OPTIONS ?= "crd:trivialVersions=true"
+```
+
+With:
+
+```
+CRD_OPTIONS ?= "crd:trivialVersions=true,preserveUnknownFields=false"
+```
+
+To allow downloading the newer versions of the K8S binaries required by EnvTest in the `bin/` directory of your project instead of use the globally setup, replace:
+
+```
+# Run tests
+test: generate fmt vet manifests
+	go test ./... -coverprofile cover.out
+```
+
+With:
+
+```
+# Run tests
+ENVTEST_ASSETS_DIR=$(shell pwd)/testbin
+test: generate fmt vet manifests
+	mkdir -p ${ENVTEST_ASSETS_DIR}
+	test -f ${ENVTEST_ASSETS_DIR}/setup-envtest.sh || curl -sSLo ${ENVTEST_ASSETS_DIR}/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/v0.7.0/hack/setup-envtest.sh
+	source ${ENVTEST_ASSETS_DIR}/setup-envtest.sh; fetch_envtest_tools $(ENVTEST_ASSETS_DIR); setup_envtest_env $(ENVTEST_ASSETS_DIR); go test ./... -coverprofile cover.out
+```
+
+<aside class="note warning">
+<h1>Note</h1>
+
+The Kubernetes binaries that are required for the EnvTest were upgraded from `1.16.4` to `1.19.2`. You can still install them globally by running: 
+
+```
+os=$(go env GOOS)
+arch=$(go env GOARCH)
+
+# download kubebuilder and extract it to tmp
+curl -LO https://storage.googleapis.com/kubebuilder-tools/kubebuilder-tools-1.19.2-${os}-${arch}.tar.gz | tar -xz -C /tmp/
+
+# move to a long-term location and put it on your path
+# (you'll need to set the KUBEBUILDER_ASSETS env var if you put it somewhere else)
+sudo mv /tmp/kubebuilder-tools-1.19.2-${os}-${arch}.tar.gz /usr/local/kubebuilder
+export PATH=$PATH:/usr/local/kubebuilder/bin
+```
+
+</aside>
+
+- Add the following new target to help you undeploy your projects: 
+
+```
+# UnDeploy controller from the configured Kubernetes cluster in ~/.kube/config
+undeploy:
+	$(KUSTOMIZE) build config/default | kubectl delete -f -
+```
+
+- Update the `docker-build` target to allow you work easily with podman since Podman prefers options before positional arguments, replace:
```

Comment:
```suggestion
- Update the `docker-build` target to allow you to work easily with podman since Podman prefers options before positional arguments, replace:
```